types/index.d.ts

```diff
@@ -1,7 +1,7 @@
 declare class OrderedMap<T = any> {
-  constructor(content: T[])
```

Comment:
Why `string | T`?

Reply:
@marijnh here's my thinking, apologies if I'm wrong.

If you say your OrderedMap contains numbers `OrderedMap<number>` you then cannot initialize it with keys `new OrderedMap<number>([ 'key1', 1, 'key2', 2])` without type errors.

To get around this, you might try `OrderedMap<string | number>` but that means ever time you do `.get('key1')` TypeScript thinks the result is number or string, so you need to do additional work to narrow the type

I added this as convince so you can pass keys in the constructor, without the need for narrowing later on

Reply:
As an aside, if the constructor took an array of arrays (like `Object.entries` gives) this would be easier to type.

i.e. `new OrderedMap([ ['key1', 1], ['key2, 2] ])`

But obviously that would be a breaking change!!

Reply:
Ah, right, should have read a bit deeper. The thing is that only the documented part of the API is public, so this could just be `private constructor()`

Reply:
Yep makes sense. Thanks, will update.vc